src/main.js

```diff
@@ -10,20 +10,99 @@ Config.set({
+    const passwordCallbacks = step.getCallbacksOfType('PasswordCallback');
+    const choiceCallbacks = step.getCallbacksOfType('ChoiceCallback');
+    
+    if (nameCallbacks.length && passwordCallbacks.length) {
+        stage = 'UsernamePassword';
+    } else if (choiceCallbacks.length) {
+        stage = 'Choice';
+    } else {
+        stage = undefined;
+    }
+    
+    console.log('stage', stage);
+    return stage;
+}
+
+function renderChoices(step) {
+    const choiceCallback = step.getCallbackOfType('ChoiceCallback');
+    // console.log('choiceCallback', choiceCallback);
+    const prompt = choiceCallback.getPrompt();
+    const choices = choiceCallback.getChoices();
+    const defaultChoice = choiceCallback.getDefaultChoice();
+    // console.log('choice', prompt, choices);
+
+    if (!prompt || !choices.length) {
+        throw new Error('No prompt or choices found');
+    }
+
+    const promptElem = document.createElement('p');
+    promptElem.innerHTML = prompt;
+    continueElem.appendChild(promptElem);
+
+    choices.forEach(choice => {
+        const radio = document.createElement('div');
+        const input = document.createElement('input');
+        input.setAttribute('type', 'radio');
+        input.setAttribute('id', choice);
+        input.setAttribute('name', 'choice');
+        input.setAttribute('value', choice);
+        input.setAttribute('checked', defaultChoice === choice);
+        radio.appendChild(input);
+
+        const label = document.createElement('label');
+        label.setAttribute('for', choice);
+        label.innerHTML = choice;
+        radio.appendChild(label);
+
+        continueElem.appendChild(radio);
+        continueElem.addEventListener('submit', () => submitChoiceHandler(step));
+    });
+
+    const submitChoiceButton = document.createElement('button');
+    submitChoiceButton.setAttribute('type', 'submit');
+    submitChoiceButton.innerHTML = 'Submit';
+    continueElem.appendChild(submitChoiceButton);
+
+    continueElem.style.display = 'block';
+}
+
 function handleFirstStep(step) {
-    const formElem = document.getElementById('login-form');
+    const loginElem = document.getElementById('login-form');
     const nameCallback = step.getCallbackOfType('NameCallback');
     const passwordCallback = step.getCallbackOfType('PasswordCallback');
-    nameCallback.setName(formElem.querySelector('input[name="username"]').value);
-    passwordCallback.setPassword(formElem.querySelector('input[name="password"]').value);
+    nameCallback.setName(loginElem.querySelector('input[name="username"]').value);
+    passwordCallback.setPassword(loginElem.querySelector('input[name="password"]').value);
     console.log('nameCallback', nameCallback);
     console.log('passwordCallback', passwordCallback);
 }
 
-async function handleSecondStep(step) {
+async function handleNextStep(step) {
+    continueElem.style.display = 'none';
     switch (step.type) {
     case 'LoginSuccess':
```

Comment:
This is definitely subjective, but I usually advocate for if statements when they contain a large blocks of logic. Switch statements are great for mapping simple logic or function calls to a particular condition.